modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc

```diff
@@ -5890,6 +5890,118 @@ Total database accesses: 5, total allocated memory: 1488
 
 ======
 
+[role=label--new-5.21]
+[[query-plan-stateful-shortest-path-into]]
+== StatefulShortestPath(Into)
+// StatefulShortestPath(Into)
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE CONSTRAINT person_name_unique IF NOT EXISTS FOR (p:Person) REQUIRE (p.name) IS UNIQUE
+----
+////
+
+The `StatefulShortestPath(Into)` operator finds shortest paths between two previously matched node variables.
+It uses a bidirectional breadth-first search (BFS) algorithm, which performs two BFS invocations at the same time, one from the left boundary node and one from the right boundary node.
+Once a node is found by both BFS invocations, which indicates that it can be reached from both boundary nodes, the algorithm successfully terminates.
+When both BFS invocations terminate without any overlap in the nodes found, then there exists no valid path between both boundary nodes.
```

Comment:
Alternative:
```suggestion
If one of the BFS invocations exhausts its search before intersecting, either because no further nodes can be reached or because the maximum number of hops has been reached, then there is no valid path between the boundary nodes and the algorithm terminates.
```

Reply:
@WilcoNeo ^^^ alternative wording.